conans/model/new_build_info.py

```diff
@@ -70,10 +74,10 @@ def get_init(self, attribute, default):
 
 class NewCppInfo(object):
 
-    def __init__(self):
-        self.components = DefaultOrderedDict(lambda: _NewComponent())
+    def __init__(self, with_defaults=False):
```

Comment:
It is not clear who is calling ``NewCppInfo(with_defaults=True)`` to change this default behavior.

Reply:
https://github.com/conan-io/conan/pull/11471/files#diff-f33d0f25da77d5b8727e9c2e931247512829f64f3ac1e688ae7aa35e0a06e39cR11